brain/compartmentReportView.cpp

```diff
@@ -120,9 +56,22 @@ const CompartmentReportMapping& CompartmentReportView::getMapping() const
 }
 
 std::future<CompartmentReportFrame> CompartmentReportView::load(
-    const float timestamp)
+    double timestamp)
 {
+    const double start = _impl->report->getStartTime();
+    const double end = _impl->report->getEndTime();
+
+    if (timestamp < start || timestamp >= end)
+        throw std::logic_error("Invalid timestamp");
+
+    const double timestep = _impl->report->getTimestep();
+
+    size_t frameIndex = std::llround((timestamp - start) / timestep);
```

Comment:
Fine with me, but this should be checked with @eile.
At the moment all caching is temporary and the compartment data caching is not used in production.